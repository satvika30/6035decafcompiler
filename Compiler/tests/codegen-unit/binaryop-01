class Program {
    void main() {
   		int a, b, c, d, e ,f;
   		
        a = 2 * 4;
        callout("printf", "%d\n", a);
        
        a = a * 2 / 4;
        callout("printf", "%d\n", a);
        
        b = 7;        
        c = a * (b / 2);
        callout("printf", "%d\n", c);
        
        f = 100 / 11 * b; 
        callout("printf", "%d\n", f);
        
        e = 100 / -11 * b;
        callout("printf", "%d\n", e);
        
        e = 100 / -11 * -b;
        callout("printf", "%d\n", e);
        
        a = 100 % 7 * 2;
        callout("printf", "%d\n", a);
    }
}
