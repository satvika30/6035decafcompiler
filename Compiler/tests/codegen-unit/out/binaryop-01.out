8
4
12
63
-63
63
4